sprocs/assessment_instances_insert.sql

```diff
@@ -26,19 +28,61 @@ BEGIN
 
     -- ######################################################################
     -- determine the "number" of the new assessment instance
+    IF group_work THEN
+        SELECT gu.group_id
+        INTO  groupId
+        FROM  group_users as gu
+        JOIN  groups AS g ON(g.id = gu.group_id)
+        JOIN  group_configs AS gc ON (gc.id = g.group_config_id)
+        WHERE gu.user_id = assessment_instances_insert.user_id AND
+        gc.assessment_id = assessment_instances_insert.assessment_id AND
+        gc.deleted_at IS NULL AND
+        g.deleted_at IS NULL;
+        
+        IF assessment.multiple_instance THEN
+            SELECT coalesce(max(ai.number), 0) + 1
+            INTO number
+            FROM assessment_instances AS ai
+            WHERE
+                ai.assessment_id = assessment_instances_insert.assessment_id
+                AND ai.group_id = groupId;
```

Comment:
Rather than duplicating this code, can we instead keep just one copy but change the WHERE statement to be:
```
AND (WHEN group_work THEN ai.group_id = groupId ELSE ai.user_id = assessment_instances_insert.user_id END)
```